src/modules/pmc/src/lib/ExecutionState.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <string>
+
+enum StateComponent
+{
+    Unknown = 0,
+    Running,
+    Succeeded,
+    Failed,
+    TimedOut
+};
+
+enum SubStateComponent
+{
+    None,
+    DeserializingJsonPayload,
+    DeserializingDesiredState,
+    DeserializingPackages,
+    UpdatingPackagesLists,
+    InstallingPackages
+};

```

Comment:
Can these be made internal to the ExecutionState class or a parent namespace be added? Just to not have the enums as global and risk to collide with others.